service/src/main/java/bio/terra/tanagra/service/accesscontrol/model/impl/SamGroupsAccessControl.java

```diff
@@ -0,0 +1,116 @@
+import bio.terra.tanagra.service.accesscontrol.AccessControlHelper;
+import bio.terra.tanagra.service.accesscontrol.Permissions;
+import bio.terra.tanagra.service.accesscontrol.ResourceCollection;
+import bio.terra.tanagra.service.accesscontrol.ResourceId;
+import bio.terra.tanagra.service.accesscontrol.ResourceType;
+import bio.terra.tanagra.service.accesscontrol.model.UnderlayAccessControl;
+import bio.terra.tanagra.service.authentication.UserId;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+
+public class SamGroupsAccessControl implements UnderlayAccessControl {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SamGroupsAccessControl.class);
+  private String basePath;
+  private SamService samService;
+  private final Map<String, String> underlayToGroup = new HashMap<>();
+
+  @Override
+  public String getDescription() {
+    return "SamGroups underlay-based access control";
+  }
+
+  @Override
+  public void initialize(
+      AccessControlConfiguration accessControlConfiguration,
+      AccessControlHelper accessControlHelper) {
+    this.basePath = accessControlConfiguration.getBasePath();
+    Assert.notNull(basePath, "Base URL is required for Sam Service API calls");
+
+    this.samService = samService;
+    // lookupGroupIds();
+  }
+
+  /*
+   private void lookupGroupIds() {
+     // Get list of all underlays
+     ResourceCollection resourceCollection = ResourceCollection.allResourcesAllPermissions(ResourceType.UNDERLAY, null);
+
+
+
+     List<?> listUnderlays = underlayService.listUnderlays(ResourceCollection.allResourcesAllPermissions())
+
+
+
+     listUnderlays
+     if (allAccessGroup != null || !underlayToGroup.isEmpty()) {
+       return; // Already initialized.
+     }
+     try {
+       // Lookup the group ID for each group name. The ID is required to list the members.
+       Map<String, VerilyGroup> nameToGroup =
+           apiListGroups().stream()
+               .collect(Collectors.toMap(VerilyGroup::getName, Function.identity()));
+
+       // Build a map of underlay name to VerilyGroup object.
+       for (int i = 0; i < params.size(); i += 2) {
+         String underlay = params.get(i);
+         String groupName = params.get(i + 1);
+         if (ALL_ACCESS.equals(underlay)) {
+           allAccessGroup = nameToGroup.get(groupName);
+         } else {
+           underlayToGroup.put(underlay, nameToGroup.get(groupName));
+         }
+       }
+     } catch (SystemException | InvalidCredentialsException ex) {
+       LOGGER.error("Error initializing VerilyGroups access control implementation", ex);
+     }
+   }
+
+  */
+
+  @Override
+  public ResourceCollection listUnderlays(UserId user, int offset, int limit) {
+    // TODO: Call sam to get all underlay resources that user has READ access to
```

Comment:
@marikomedlock fyi - the TODO comments capture the logic. 
implementation is placeholder / in progress